highfidelity/hifi
# Case 20882: Support custom display / input plugin lists and small tweaks

[20882](https://highfidelity.manuscript.com/f/cases/20882/Support-CMake-time-overrides-of-normal-display-plugins)

* Updates the CMake configuration and the `input-plugins` and `display-plugins` libraries to allow applications to suppress the default built-in plugins and supply their own (necessary for Oculus Mobile and potential other future standalone HMDs)
* Expose some critical OpenGL ES plugin availability to the code via preprocessor macros
* Ensure GLES backends will use the instanced stereo drawing technique if `EXT_clip_cull_distance` is available
* Merge in updates to the `oculusMobile` and `oculusMobilePlugin` libraries from the feature/quest branch
* Ensure the QML file selector is exposed at the `hifi::qml::OffscreenSurface` level
* Remove use of `FileLogger` in android builds (only logcat should be used, not file based logging)

## Testing

* Verify file logging works correctly on desktop builds.
* Verify Android startup and shutdown work the same as in master builds
* Verify Android rendering is correct
* Verify the list of display plugins is unchanged from master
* Verify web content (web entities, marketplace) works from Desktop builds.
